NewsFeed/Common/Views/ScrollableSegmentedControl.swift

```diff
@@ -0,0 +1,235 @@
+//
+//  ScrollableSegmentedControl.swift
+//  ScrollableSegmentedControl
+//
+//  Created by MArko Satlan on 22/07/2018.
+//  Copyright © 2018 MArko Satlan. All rights reserved.
+//
+
+import UIKit
+
+class ScrollableSegmentedControl: UIControl {
+
+// MARK: - Properties
+    // Open
+    open var font = UIFont(name: "Georgia", size: 15) {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var buttonPadding: CGFloat = 15 {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var buttonMargin: CGFloat = 15 {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var selectedSegmentIndex: Int = -1 {
+        didSet {
+            highlightSelectedSegment()
+            calculateAndSetScrollOffset()
+            
+            // deselect previous button if different button is tapped
+            if oldValue != selectedSegmentIndex {
+                let previousButton = buttonsArray.filter{$0.tag == oldValue}.first
+                previousButton?.isSelected = false
+                UIButton.animate(withDuration: 0.3, animations: {
+                    previousButton?.transform = CGAffineTransform.identity
+                })
+            }
+        }
+    }
+    
+    // Private
+    private var buttonTitles: [String] = []
+    private var buttonsArray: [UIButton] = []
+    private let scrollView = UIScrollView()
+    
+// MARK: - Methods
+    // Init
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        
+        scrollView.showsHorizontalScrollIndicator = false
+        addSubview(scrollView)
+    }
+    
+    public init(items: [String]) {
+        self.buttonTitles = items
+        
+        super.init(frame: CGRect.zero)
+        
+        scrollView.showsHorizontalScrollIndicator = false
+        addSubview(scrollView)
```

Comment:
Those two lines are the same as in `init?(coder aDecoder: NSCoder)` method. I suggest creating a method such as `private func setUp()` and call that from both `init` methods.